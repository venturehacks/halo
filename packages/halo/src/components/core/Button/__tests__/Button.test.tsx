import { mount, render } from 'enzyme';
import React from 'react';

import { Button } from '../index';

describe('Button', () => {
  test('smoke', () => {
    const component = render(<Button>PHP</Button>);
    expect(component.text()).toEqual('PHP');
  });

  describe('dismiss', () => {
    test('callback executes', () => {
      const dismiss = jest.fn();
      const wrapper = mount(<Button onClick={dismiss} />);

      expect(dismiss).toBeCalledTimes(0);

      wrapper.find(`[data-test="Button"]`).last().simulate('click');

      expect(dismiss).toBeCalledTimes(1);
    });
  });

  describe('snapshot', () => {
    test('default', () => {
      const component = render(<Button />);
      expect(component).toMatchSnapshot();
    });

    test('width 100%', () => {
      const component = render(<Button width="100%" />);
      expect(component).toMatchSnapshot();
    });

    describe('size', () => {
      test('sm', () => {
        const component = render(<Button size="sm" />);
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(<Button size="md" />);
        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = render(<Button size="lg" />);
        expect(component).toMatchSnapshot();
      });
    });

    describe('variant', () => {
      test('primary', () => {
        const component = render(<Button variant="primary" />);
        expect(component).toMatchSnapshot();
      });

      test('secondary', () => {
        const component = render(<Button variant="secondary" />);
        expect(component).toMatchSnapshot();
      });

      test('gray', () => {
        const component = render(<Button variant="gray" />);
        expect(component).toMatchSnapshot();
      });

      test('error', () => {
        const component = render(<Button variant="error" />);
        expect(component).toMatchSnapshot();
      });

      test('clear', () => {
        const component = render(<Button variant="clear" />);
        expect(component).toMatchSnapshot();
      });
    });

    describe('gtm variant', () => {
      test('gtm-primary', () => {
        const component = render(<Button variant="gtm-primary" />);
        expect(component).toMatchSnapshot();
      });

      test('gtm-secondary', () => {
        const component = render(<Button variant="gtm-secondary" />);
        expect(component).toMatchSnapshot();
      });

      test('gtm-gray', () => {
        const component = render(<Button variant="gtm-gray" />);
        expect(component).toMatchSnapshot();
      });

      test('gtm-error', () => {
        const component = render(<Button variant="gtm-error" />);
        expect(component).toMatchSnapshot();
      });

      test('gtm-clear', () => {
        const component = render(<Button variant="gtm-clear" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
