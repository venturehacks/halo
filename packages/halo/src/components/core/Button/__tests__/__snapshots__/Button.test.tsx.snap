// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot default 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot gtm variant gtm-clear 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-transparent font-normal text-black
      disabled:text-gray-600 disabled:font-medium hover:font-bold"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot gtm variant gtm-error 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-gtmpink-100 border-gtmpink-100 text-gtmpink-600 font-formal
      disabled:border-gray-200 disabled:text-gray-700 disabled:bg-gray-200
      hover:bg-gtmpink-300 hover:border-gtmpink-300 hover:text-gtmpink-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot gtm variant gtm-gray 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-gray-200 border-black text-black font-normal hover:bg-gray-400
      disabled:bg-gray-100 disabled:border-gray-500 disabled:text-gray-500"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot gtm variant gtm-primary 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-black border-black text-white font-normal
      disabled:bg-gray-400 disabled:border-gray-400 disabled:text-gray-700
      hover:bg-gtmblue-500 hover:border-gtmblue-500"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot gtm variant gtm-secondary 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-white border-black text-black font-normal
      disabled:bg-white disabled:border-gray-700 disabled:text-gray-700
      hover:bg-gtmblue-100 hover:border-gtmblue-500 hover:text-gtmblue-500"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot size lg 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-10 h-12 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot size md 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-4 h-10 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot size sm 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot variant clear 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-transparent font-medium text-dark-aaa
      disabled:text-dark-a disabled:font-medium hover:font-bold"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot variant error 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-red-100 border-red-100 text-light-error font-medium
      disabled:border-gray-100 disabled:text-gray-500 disabled:bg-gray-100
      hover:bg-red-200"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot variant gray 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-gray-100 border-dark-aaa text-dark-aaa font-medium
      disabled:bg-gray-100 disabled:border-dark-a disabled:text-dark-a
      hover:bg-slate-200"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot variant primary 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot variant secondary 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-white border-blue-500 text-blue-500 font-medium
      disabled:bg-white disabled:border-blue-300 disabled:text-blue-300
      hover:bg-slate-100"
  data-test="Button"
  type="button"
/>
`;

exports[`Button snapshot width 100% 1`] = `
<button
  class="rounded border-solid border-transparent border gap-x-2.5 mr-4 mb-2 whitespace-nowrap
      antialiased text-center text-sm no-underline cursor-poinnter focus:outline-0
      disabled:cursor-default disabled:pointer-events-none disabled:opacity-50 px-3 h-8 bg-dark-brand border-dark-brand text-white font-medium
      disabled:bg-blue-300 disabled:border-blue-300
      hover:bg-blue-600 hover:border-blue-600 w-full"
  data-test="Button"
  type="button"
/>
`;
